migrations/sql/V2020.03.12.08.33__create_bond_structure.sql

```diff
@@ -0,0 +1,56 @@
+    active_ind                       boolean                  DEFAULT true  NOT NULL            ,
+    create_user                      varchar                                NOT NULL            ,
+    create_timestamp                 timestamp with time zone DEFAULT now() NOT NULL            ,
+    update_user                      varchar                                NOT NULL            ,
+    update_timestamp                 timestamp with time zone DEFAULT now() NOT NULL
+);
+
+CREATE TABLE bond_type_code (
+    bond_type_code                   varchar                                NOT NULL PRIMARY KEY,
+    description                      varchar                                NOT NULL            ,
+    active_ind                       boolean                  DEFAULT true  NOT NULL            ,
+    create_user                      varchar                                NOT NULL            ,
+    create_timestamp                 timestamp with time zone DEFAULT now() NOT NULL            ,
+    update_user                      varchar                                NOT NULL            ,
+    update_timestamp                 timestamp with time zone DEFAULT now() NOT NULL
+);
+
+ALTER TABLE bond_status_code OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS bond (
+    bond_id                                                          SERIAL PRIMARY KEY,
+    bond_guid                        uuid DEFAULT gen_random_uuid()            NOT NULL,
+    amount                           numeric(14,2)                             NOT NULL,
+    bond_type_code                   varchar                                   NOT NULL,
+    payer_party_guid                 uuid                                      NOT NULL,
+    institution_party_guid           uuid                                              ,
+    bond_status_code                 varchar                                   NOT NULL,
+    reference_number                 varchar                                           ,
+    create_user                      varchar                                   NOT NULL,
+    create_timestamp                 timestamp with time zone DEFAULT now()    NOT NULL,
+    update_user                      varchar                                   NOT NULL,
+    update_timestamp                 timestamp with time zone DEFAULT now()    NOT NULL,
+
+    FOREIGN KEY (payer_party_guid) REFERENCES party(party_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (institution_party_guid) REFERENCES party(party_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (bond_status_code) REFERENCES bond_status_code(bond_status_code) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (bond_type_code) REFERENCES bond_type_code(bond_type_code) DEFERRABLE INITIALLY DEFERRED
+
+);
+
+ALTER TABLE bond OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS bond_permit_xref
+(
+    bond_permit_xref_guid uuid    DEFAULT gen_random_uuid() NOT NULL PRIMARY KEY,
```

Comment:
The XREF GUID isn't needed if we use a primary key of (bond_id, permit_id). I thought we wanted to ditch serial keys on XREF tables but maybe it doesn't matter so much anymore.

Reply:
I don't remember that conversation or choice but am more than happy to.